sim/dex-data.ts

```diff
@@ -905,70 +911,72 @@ export class Move extends BasicEffect implements Readonly<BasicEffect & MoveData
 		this.stab = data.stab || undefined;
 		this.volatileStatus = typeof data.volatileStatus === 'string' ? (data.volatileStatus as ID) : undefined;
 
-		if (this.category !== 'Status' && !this.gmaxPower) {
+		if (this.category !== 'Status' && !this.maxMove?.basePower) {
+			this.maxMove = this.maxMove || {} as MoveData['maxMove'];
 			if (this.isMax || this.isZ) {
-				this.gmaxPower = 1;
+				this.maxMove!.basePower = 1;
 			} else if (!this.basePower) {
-				this.gmaxPower = 100;
+				this.maxMove!.basePower = 100;
 			} else if (['Fighting', 'Poison'].includes(this.type)) {
 				if (this.basePower >= 150) {
-					this.gmaxPower = 100;
+					this.maxMove!.basePower = 100;
 				} else if (this.basePower >= 110) {
-					this.gmaxPower = 95;
+					this.maxMove!.basePower = 95;
 				} else if (this.basePower >= 75) {
-					this.gmaxPower = 90;
+					this.maxMove!.basePower = 90;
 				} else if (this.basePower >= 65) {
-					this.gmaxPower = 85;
+					this.maxMove!.basePower = 85;
 				} else if (this.basePower >= 55) {
-					this.gmaxPower = 80;
+					this.maxMove!.basePower = 80;
 				} else if (this.basePower >= 45) {
-					this.gmaxPower = 75;
+					this.maxMove!.basePower = 75;
 				} else {
-					this.gmaxPower = 70;
+					this.maxMove!.basePower = 70;
 				}
 			} else {
 				if (this.basePower >= 150) {
-					this.gmaxPower = 150;
+					this.maxMove!.basePower = 150;
 				} else if (this.basePower >= 110) {
-					this.gmaxPower = 140;
+					this.maxMove!.basePower = 140;
 				} else if (this.basePower >= 75) {
-					this.gmaxPower = 130;
+					this.maxMove!.basePower = 130;
 				} else if (this.basePower >= 65) {
-					this.gmaxPower = 120;
+					this.maxMove!.basePower = 120;
 				} else if (this.basePower >= 55) {
-					this.gmaxPower = 110;
+					this.maxMove!.basePower = 110;
 				} else if (this.basePower >= 45) {
-					this.gmaxPower = 100;
+					this.maxMove!.basePower = 100;
 				} else {
-					this.gmaxPower = 90;
+					this.maxMove!.basePower = 90;
 				}
 			}
 		}
-		if (this.category !== 'Status' && !this.zMovePower && !this.isZ && !this.isMax) {
+		if (this.category !== 'Status' && !this.zMove?.basePower && !this.isZ && !this.isMax) {
 			let basePower = this.basePower;
+			this.zMove = this.zMove || {} as MoveData['zMove'];
```

Comment:
```suggestion
			this.zMove = {};
```

Reply:
(With the `!this.zMove` change, there's no need for this to be more complicated.)